### PR TITLE
update bazel to v2.0.0

### PR DESCRIPTION
**Pull request by rluria14**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by rluria14** [comment]
@conda-forge-admin, please re-render

**Comment by rluria14** [comment]
@conda-forge-admin, please re-render

**Comment by rluria14** [comment]
@jjhelmus @nehaljwani @abhi18av please review when you have a chance!

**Comment by nehaljwani** [comment]
@jjhelmus Has Anaconda Distribution moved to 10.12?

**Comment by jjhelmus** [comment]
> Has Anaconda Distribution moved to 10.12?

We have not, our SDK target is 10.10 with a few cases where we still use 10.9

**Comment by rluria14** [comment]
Understood but it seems that bazel can only be built with 10.12 on osx. I've tried to override with the conda_build_config.yaml but it's not working.

**Comment by rluria14** [comment]
It seems that bazel can only be built with 10.12 SDK target on OSX. I've included the conda_build_config.yaml to set macosx_deployment_target to 10.12, re-rendered, and now it's passing. 

Also, for reference, it looks like [qt-feedstock ](https://github.com/conda-forge/qt-feedstock) uses the 10.12 SDK.

@jjhelmus @nehaljwani

**Comment by rluria14** [comment]
@jjhelmus @nehaljwani bump

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,29 +1,28 @@
...
-    - fix_path.patch  # [linux]
+  sha256: 724da3c656f68e787a86ebb9844773aa1c2e3a873cc39462a8f1b336153d6cbb
 
 build:
   number: 0
   ignore_prefix_files: True
   binary_relocation: False
   skip: True  # [win]
+  missing_dso_whitelist:
+    - '*'
```
Comment (jjhelmus) [inline recipe/meta.yaml]: Can this be made more explicit?  White listing everything (like is done here) essentially skips the under/overlinking check.

**Comment by jjhelmus** [comment]
I'm not certain if this will work on CentOS 6 given that binaries embedded in the bazel self-extracting zip are dynamically linked to the conda provided libstdc++ and when unpacked will not have a correct RPATH.  Still worth trying.

**Comment by rluria14** [comment]
@jjhelmus I do seem to be having a problem with the RPATH and libstdc++. 

Errors in build:

`bazel: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.15' not found (required by bazel)
bazel: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by bazel)
bazel: /usr/lib64/libstdc++.so.6: version `CXXABI_1.3.5' not found (required by bazel)
bazel: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.22' not found (required by bazel)
bazel: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by bazel)
bazel: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.14' not found (required by bazel)
bazel: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.18' not found (required by bazel)
bazel: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.19' not found (required by bazel)
`
Any suggestions on how to fix this?

**Comment by jjhelmus** [comment]
> Any suggestions on how to fix this?

In the recipe we use a Anaconda [we statically link libstdc++ and libgcc](https://github.com/AnacondaRecipes/bazel-feedstock/blob/2b251766c8397538833bfd7c0de2b88c03fed785/recipe/build.sh#L32).  Unfortunately I have been able to get this to work on bazel >=0.27.

**Pull request merged**